src/nls.jl

```diff
@@ -148,23 +149,23 @@ function NLPModels.jprod_residual!(nls :: ADNLSModel, x :: AbstractVector, v ::
   @lencheck nls.meta.nvar x v
   @lencheck nls.nls_meta.nequ Jv
   increment!(nls, :neval_jprod_residual)
-  Jv .= ForwardDiff.derivative(t -> nls.F(x + t * v), 0)
+  Jv .= pushforward(nls.adbackend, nls.F, x, v)
   return Jv
 end
 
 function NLPModels.jtprod_residual!(nls :: ADNLSModel, x :: AbstractVector, v :: AbstractVector, Jtv :: AbstractVector)
   @lencheck nls.meta.nvar x Jtv
   @lencheck nls.nls_meta.nequ v
   increment!(nls, :neval_jtprod_residual)
-  Jtv .= ForwardDiff.gradient(x -> dot(nls.F(x), v), x)
+  Jtv .= pullback(nls.adbackend, nls.F, x, v)
   return Jtv
 end
 
 function NLPModels.hess_residual(nls :: ADNLSModel, x :: AbstractVector, v :: AbstractVector)
   @lencheck nls.meta.nvar x
   @lencheck nls.nls_meta.nequ v
   increment!(nls, :neval_hess_residual)
-  return tril(ForwardDiff.jacobian(x -> ForwardDiff.jacobian(nls.F, x)' * v, x))
+  return tril(jacobian(nls.adbackend, x -> pullback(nls.adbackend, nls.F, x, v), x))
```

Comment:
You mentioned that `ZygoteAD` fails on the consistency checks, so you removed it. It looks like it's because of this line and the other `hess_coord_residual` below. I changed to the following:
```suggestion
  ϕ(x) = dot(nls.F(x), v)
  return tril(hessian(nls.adbackend, ϕ, x))
```
as similarly to `hess_coord_residual`.
I don't know enough about ADs to know if this breaks something else.

Reply:
ya this was it, thanks!